### PR TITLE
Pf4 rebased

### PR DESCRIPTION
**Pull request by secondsun**
## Motivation
This is THE Patternfly 4 rewrite PR, plus a couple of small bug fixes rebased on master.

Good luck.

### CODE REVIEW

**Comment by secondsun** [comment]
@ziccardi I propose we allow the merge even with the commit lint failures. This is a PR for you to check, I found a couple of problems with the sender UI that I fixed, as well as a rebase on master. Ideally nothing is more broken than it was before I rebased.

File: src/application/crud/DeleteApplicationPage.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        this.props.close();
+      }
+    } catch (err) {
+      (this.context as ContextInterface).alert(err);
+    }
+  };
+
+  render(): React.ReactNode {
+    return this.props.app ? (
+      <Modal
```
Comment (ziccardi) [inline src/application/crud/DeleteApplicationPage.tsx:53]: I\d change this to a simple:
```typescript
if (!this.props.app) {
   return <></>;
}
return ( <Modal
    ...
```

File: src/application/crud/DeleteVariantPage.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    variant: Variant,
+    providedName: string
+  ) => {
+    try {
+      if (providedName === variant.name) {
+        await UpsClientFactory.getUpsClient()
+          .variants.delete(app.pushApplicationID)
+          .withName(providedName)
+          .execute();
+        // this.props.app.variants = this.props.app!.variants!.filter(filteredVar => filteredVar.variantID !== variant.variantID);
```
Comment (ziccardi) [inline src/application/crud/DeleteVariantPage.tsx:46]: delete

File: src/application/crud/DeleteVariantPage.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        this.props.close();
+        (this.context as ContextInterface).refresh();
+      }
+    } catch (err) {
+      console.log(err);
```
Comment (ziccardi) [inline src/application/crud/DeleteVariantPage.tsx:51]: replace with `(this.context as ContextInterface).alert(err);`

File: src/application/crud/RenameVariantPage.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  ) => {
+    try {
+      await UpsClientFactory.getUpsClient()
+        .variants[variant.type].update(app.pushApplicationID, variant.variantID)
+        .withName(providedName)
+        .execute();
+      this.props.close();
+      (this.context as ContextInterface).refresh();
+    } catch (err) {
+      console.log(err);
```
Comment (ziccardi) [inline src/application/crud/RenameVariantPage.tsx:48]: Replace with `(this.context as ContextInterface).alert(err);`

File: src/application/wizard/SendTestNotificationPage.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+          </Split>
+          {variantReady ? variantReadyContent : buildVariantContent}
+        </TextContent>
+      </Page>
+      <Text component={TextVariants.p}>
+        <b>Message</b>
+      </Text>
+      <TextArea
+        disabled={!variantReady}
+        defaultValue={`Hello! This is my first notification to ${props.variant.name}`}
```
Comment (ziccardi) [inline src/application/wizard/SendTestNotificationPage.tsx:172]:
```suggestion
        defaultValue={testMessage}
```

File: startup.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+
+destfile="/usr/share/nginx/html/config.txt"
+echo "$UPS_HOST" > "$destfile"
+nginx -g "daemon off;"
```
Comment (ziccardi) [inline startup.sh:7]: this is both here and in the docker file as a CMD.
Let's take just one

**Review by ziccardi — CHANGES_REQUESTED** [review]

File: src/application/ApplicationDetail/panels/ActivityLogPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+              this.setState({ currentPage: 1 });
+              this.loadActivityLog(0);
+            }}
+            onLastClick={() => {
+              this.setState({ currentPage: Math.ceil(this.state.total / 10) });
+              this.loadActivityLog(Math.ceil(this.state.total / 10) - 1);
+            }}
+            onPerPageSelect={(event, perPage, currentPage) => {
+              this.setState({ currentPage, perPage }, () =>
+                this.loadActivityLog(currentPage - 1)
```
[the remark this answers is not in the document]
Comment (secondsun) [reply src/application/ApplicationDetail/panels/ActivityLogPanel.tsx:250]: The loadActivityLog is 0 indexed, but the pager is 1 indexed

**Review by ziccardi — APPROVED** [review]
Agree with overriding commitlint for this PR.
👍

**Pull request merged**